src/game_initialization/lobby_data.cpp

```diff
@@ -215,7 +215,7 @@ game_info::game_info(const config& game, const std::vector<std::string>& install
 	// Parse the list of addons required to join this game.
 	for(const config& addon : game.child_range("addon")) {
 		if(addon.has_attribute("id")) {
-			if(std::find(installed_addons.begin(), installed_addons.end(), addon["id"].str()) == installed_addons.end()) {
+			if(std::find(installed_addons.begin(), installed_addons.end(), addon["id"].str()) == installed_addons.end() && addon["require"].to_bool(false)) {
```

Comment:
I think you need to check it at both places, this place checks for addons with require_download that are not installed  and the other palce checks for addons with require_download that are outdated